### PR TITLE
Translate string.rst:777 to 786

### PR DESCRIPTION
**Pull request by proudofsimin**
Hi, All
Please review. Thank you.

### CODE REVIEW

File: library/string.po [hunk 1]
```diff
@@ -984,6 +982,9 @@ msgid ""
 "keywords are the placeholders.  When both *mapping* and *kwds* are given and "
 "there are duplicates, the placeholders from *kwds* take precedence."
 msgstr ""
+"進行樣本替換，並回傳一個新的字串。*mapping*是任何金鑰符合樣本裡需填入之處的字"
```
Comment (ken71301) [inline library/string.po]:
```suggestion
"進行樣本替換，並回傳一個新的字串。*mapping* 是任何金鑰符合樣本裡需填入之處的字"
```

Comment (ken71301) [reply library/string.po]: placeholders 可翻為 佔位符號
https://github.com/python/python-docs-zh-tw/blob/9187e5780fc474ab1c82e4824e9807f065a3b642/library/reprlib.po#L86

File: library/string.po [hunk 2]
```diff
@@ -984,6 +982,9 @@ msgid ""
...
+"典型物件。同時，如果關鍵字就是樣本裡需填入的物件時，您也可以改使用關鍵字引"
+"數。當*mapping*跟*kwds*同時存在並且重疊時，*kwds*會被優先使用。"
```
Comment (rockleona) [inline library/string.po]: - 空格的規則可以查閱 README 中的 rst 語法規則
- 

```suggestion
"進行樣本替換，並回傳一個新的字串。*mapping* 是任何金鑰符合樣本裡需填入之處的字"
"典型物件。同時，如果關鍵字就是樣本裡需填入的物件時，您也可以改使用關鍵字引"
"數。當 *mapping* 跟 *kwds* 同時存在並且重疊時，*kwds* 會被優先使用。"
```

**Review by rockleona — COMMENTED** [review]
Review Part 1

**Review by ken71301 — COMMENTED** [review]
用詞與空格需修正

File: library/string.po [hunk 3]
```diff
@@ -984,6 +982,9 @@ msgid ""
 "keywords are the placeholders.  When both *mapping* and *kwds* are given and "
 "there are duplicates, the placeholders from *kwds* take precedence."
 msgstr ""
+"進行樣本替換，並回傳一個新的字串。 *mapping* 是任何有金鑰並符合樣本裡佔位符號"
+"的字典型物件。同時，如果關鍵字就是佔位符號時，您也可以改使用關鍵字引數。當 "
```
Comment (ken71301) [inline library/string.po]: `alternatively` 的意思應該接近於 `或者` 而非 `同時`
原義：**used to suggest another possibility**

`provide` 應該是 `提供` 而非 `使用`

File: library/string.po [hunk 4]
```diff
@@ -984,6 +982,9 @@ msgid ""
...
+"*mapping* 跟 *kwds* 同時存在並且重疊時，*kwds* 會被優先使用。"
```
Comment (ken71301) [inline library/string.po]: `given` 應為 `給定`
這裡的duplicate意思應較接近 `重複` 而非 `重疊`
placeholders 漏翻

File: library/string.po [hunk 5]
```diff
@@ -993,6 +994,10 @@ msgid ""
 "unlike with :meth:`substitute`, any other appearances of the ``$`` will "
 "simply return ``$`` instead of raising :exc:`ValueError`."
 msgstr ""
+"與 :meth:`substitute` 類似，但如果佔位符號沒有填入任何 *mapping* 與 *kwds* 的"
```
Comment (ken71301) [inline library/string.po]: 這裡應該是 *mapping* 與 *kwds* 中缺失 placeholders

intact 為 **complete and in the original state** 所以這句應為 `會完整地出現在結果字串中`

File: library/string.po [hunk 6]
```diff
@@ -1009,6 +1014,8 @@ msgid ""
 "Returns false if the template has invalid placeholders that will cause :meth:"
 "`substitute` to raise :exc:`ValueError`."
 msgstr ""
+"當樣本裡面包含無效的佔位符號導致 :meth:`substitute` 顯示 :exc:`ValueError`，"
```
Comment (ken71301) [inline library/string.po]: raise error 的 raise 應譯為 `引發`

**Review by ken71301 — COMMENTED** [review]
仍有部分翻譯錯誤

**Comment by mattwang44** [comment]
rebase

File: library/string.po [hunk 7]
```diff
@@ -1016,6 +1019,10 @@ msgid ""
 "unlike with :meth:`substitute`, any other appearances of the ``$`` will "
 "simply return ``$`` instead of raising :exc:`ValueError`."
 msgstr ""
+"與 :meth:`substitute` 類似，但如果佔位符號沒有填入任何 *mapping* 與 *kwds* 的"
+"時候，原始的佔位符號會出現在密集的結果字串裡，而不是顯示例外 `KeyError`。同"
```
Comment (mattwang44) [inline library/string.po]:
```suggestion
"時候，原始的佔位符號會出現在密集的結果字串裡，而不是顯示例外 :exc:`KeyError`。同"
```

**Comment by rockleona** [comment]
@proudofsimin rebase 需要協助的話都可以跟我說喔
也可以到 Discord 私訊我

**Comment by proudofsimin** [comment]
> @proudofsimin rebase 需要協助的話都可以跟我說喔
> 也可以到 Discord 私訊我

Sry for the late reply. I’m pretty occupied these days. I’m not sure what that means but will try to work on it next week. Many thanks!

**Comment by proudofsimin** [comment]
Hi,
I googled what rebase means, but I'm not sure if there's anything else I should've done other than regular git check-in. Please let me know if I have use the command correctly. Thank you.

**Comment by mattwang44** [comment]
Hi @proudofsimin 
If you're referring to the `rebase` comment I left last time, it means I've done a `git rebase` ([doc](https://docs.github.com/en/get-started/using-git/about-git-rebase) and [visualization](https://dev.to/lydiahallie/cs-visualized-useful-git-commands-37p1#rebase)) for the branch of this PR. Whenever [your branch has any conflict with the target branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/about-merge-conflicts) (like now), you'll need to resolve with `git rebase ...` or `git merge ...` command. We prefer the former as the commit history would be more readable, which eases the pain for reviewers. You can [resolve the conflict on the GitHub UI](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github), but it runs a `git merge ...` command under the hood.

Feel free to pin us on the Discord server if you have any questions!

**Comment by mattwang44** [comment]
嗨 @proudofsimin 
最近要 wrap up 我們 11 月活動的所有 PR 了，想請問你有空來處理這個 PR 嗎？
還是我能否幫你來修正剩下未完成的部分呢？

File: library/string.po (hunk 6 above)
Comment (ken71301) [reply library/string.po]: fix this and we are good to go.

**Review by ken71301 — COMMENTED** [review]
fix the sentence in L:1053 and we are good to go.

**Comment by proudofsimin** [comment]
> 嗨 @proudofsimin 最近要 wrap up 我們 11 月活動的所有 PR 了，想請問你有空來處理這個 PR 嗎？ 還是我能否幫你來修正剩下未完成的部分呢？

好，我好感動謝謝！！我自己在local端試了好幾次修不好我的environment弄不上去...非常感謝

**Comment by mattwang44** [comment]
@proudofsimin 好呀，我就先處理這個 PR，如果是對於 git or github 操作上遇到問題的話都歡迎在 discord 上找我們討論歐！

**Comment by proudofsimin** [comment]
> @proudofsimin 好呀，我就先處理這個 PR，如果是對於 git or github 操作上遇到問題的話都歡迎在 discord 上找我們討論歐！

好的超級感謝，我其實之前沒用過discord所以一直都不敢發言...造成您們的困擾了非常不好意思ＱＱ

**Comment by mattwang44** [comment]
@proudofsimin 沒關係的 沒有造成困擾～
如果不敢公開發言的話也可以考慮私訊我 (my id: `mattwang44`)，不用怕麻煩到我（只是我比較忙的話就會比較晚回而已XD）

**Pull request merged**